jormungandr/src/blockchain/checkpoints.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{blockcfg::HeaderHash, blockchain::Ref};
+use std::sync::Arc;
+
+/// list of pre-computed checkpoints from a given [`Ref`].
+///
+/// [`Ref`]: ./struct.Ref.html
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Checkpoints(Vec<HeaderHash>);
+
+impl Checkpoints {
+    /// create a new list of checkpoints from the given starting point (tip).
+    ///
+    /// The function make use of the `Ref` object pointing to previous epoch in order
+    /// to populate _interestingly_ spaced checkpoints.
+    ///
+    /// For now the algorithm provide the current block, the parent, the last block of the
+    /// previous epoch, the last block of the epoch before that... until the block0.
+    pub fn new_from(from: Arc<Ref>) -> Self {
```

Comment:
This should be just `new` IMHO, no need to have strange constructor names.

Reply:
`new` would mean `empty`. here we are creating checkpoints *from* the given object.